lib/okta/service.rb

```diff
@@ -51,5 +45,26 @@ def delete_grant(user_id, grant_id)
         delete_url_with_token("#{USER_API_BASE_PATH}/#{user_id}/grants/#{grant_id}")
       end
     end
+
+    def metadata
+      with_monitoring do
+        get_url_with_token(Settings.oidc.auth_server_metadata_url)
+      end
+    end
+
+    def keys
+      url = metadata.body['jwks_uri']
+      with_monitoring do
+        get_url_with_token(url)
+      end
+    end
```

Comment:
There's lots of stuff related to "keys" when it comes to okta. Can this function be named `oidc_jwks_keys` or something like that?